py4cast/datasets/titan/__init__.py

```diff
@@ -700,7 +658,26 @@ def __getitem__(self, index: int) -> Item:
 
         # TODO: build a single NamedTensor with all the inputs and outputs in one shot.
         sample = self.sample_list[index]
-        item = sample.load()
+
+        lforcings = []
+        time_forcing = NamedTensor(  # doy : day_of_year
+            feature_names=["cos_hour", "sin_hour", "cos_doy", "sin_doy"],
+            tensor=self.get_year_hour_forcing(sample.date_t0, sample.terms).type(
+                torch.float32
+            ),
+            names=["timestep", "features"],
+        )
+        solar_forcing = NamedTensor(
+            feature_names=["toa_radiation"],
+            tensor=self.generate_toa_radiation_forcing(
+                self.grid, sample.date_t0, sample.terms
+            ).type(torch.float32),
+            names=["timestep", "lat", "lon", "features"],
+        )
+        lforcings.append(time_forcing)
+        lforcings.append(solar_forcing)
+
+        item = sample.load(lforcings)
```

Comment:
Hi ! I'd prefer if you would put the solar forcing in the `Sample.load` method, so that the Sample class contains everything it need to load.